### DISCUSSION TITLE
Is it possible to configure Celery Workers to handle multiple queues with respective concurrency?

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#42899] [Q&A] [answered]
Is it possible to configure Celery Workers to handle multiple queues with respective concurrency?

Example:

Worker 1:
`celery worker -q default,queueA,queueB`

`default` concurrency: 16
`queueA` concurrency: 4
`queueB`concurrency: 6

I would like to avoid running multiple workers per queue if possible, if not all good!

**Comment by andresitorusz** [comment, marked as answer]
AFAIK, Celery doesn't natively support assigning  different concurrency levels to each queue on the same worker out of the box.

You would typically run multiple workers with specific concurrency settings for each queue.

Check this [thread](https://github.com/celery/celery/issues/1599). It might helps.

**Reply by pedro-cf** [reply]
yes currently I adopted the multiple worker solution, but was hoping there was a better solution, thanks!
